src/middlewared/middlewared/plugins/docker/state_management.py

```diff
@@ -48,6 +48,10 @@ async def start_service(self, mount_datasets: bool = False):
             if mount_datasets:
                 catalog_sync_job = await self.middleware.call('docker.fs_manage.mount')
```

Comment:
This is confusing. Should we be waiting on `catalog_sync_job` since it's calling `docker.fs_manage.mount`? Those 2 names are very confusing and it makes me think that calling `docker.fs_manage.mount` and not waiting on it to complete before we call `docker.fs_manage.ix_apps_is_mounted` will always fail?

Reply:
No, what happens is that when we mount, a catalog sync job is triggered and we get the reference back of the job. Now the sync job has nothing to do with getting the datasets to be mounted etc and we definitely don't want to wait on it before attempting to start docker as these are unrelated and docker/apps function just fine without it (minus the ability to install new apps ofc).
The check jsut validates that the datasets are mounted as they should be

Reply:
So the endpoint `docker.fs_manage.mount` doesn't actually do any type of mounting?